crates/matrix-sdk/src/oidc.rs

```diff
@@ -452,7 +425,44 @@ impl Oidc {
             &mut rand::thread_rng(),
         )?;
 
-        let code = (use_authorization_url)(url, validation_data.state.clone()).await?;
+        let state = validation_data.state.clone();
+
+        self.client.inner.oidc_validation_data.write().await.insert(state.clone(), validation_data);
```

Comment:
It should be `lock` not `write`. Does that compile?

Reply:
My bad, I noticed this with the CLI update - I wasn't compiling with the feature enabled 🤦‍♂️